src/app/containers/ConsentBanner/CanonicalLogic/setCookieOven.test.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable global-require */
+let fetchResponse;
+let cookieOvenUrl;
+
+describe('setCookieOven', () => {
+  beforeEach(() => {
+    jest.mock('./cookieOvenUrl', () => jest.fn());
+    cookieOvenUrl = require('./cookieOvenUrl');
+    cookieOvenUrl.mockImplementation(() => 'https://cookieOvenUrl.com');
+
+    fetch.mockImplementation(() => fetchResponse);
+  });
+
+  afterEach(() => {
+    fetch.resetMocks();
+  });
+
+  it(`should fetch`, () => {
+    const setCookieOven = require('./setCookieOven').default;
+
+    setCookieOven('cookie', 'value');
+
+    expect(cookieOvenUrl).toHaveBeenCalledWith('http://localhost');
+    expect(fetch).toHaveBeenCalledWith(
+      'https://cookieOvenUrl.com/cookie/value',
+    );
+  });
+
+  describe('when the fetch fails', () => {
+    let error;
+
+    beforeEach(() => {
+      error = new Error('An error');
+      fetchResponse = Promise.reject(error);
+      global.console = { error: jest.fn(), log: console.log };
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it(`should send error to console when logger function isn't provided`, async () => {
+      const setCookieOven = require('./setCookieOven').default;
+
+      await setCookieOven('cookie', 'value');
+
+      // eslint-disable-next-line no-console
+      expect(console.error).toHaveBeenCalledWith(error);
+    });
+
+    it(`should send error to logger function when provided`, async () => {
+      const logger = { error: jest.fn() };
+
+      const setCookieOven = require('./setCookieOven').default;
+
+      await setCookieOven('cookie', 'value', logger);
+
```

Comment:
Tests look good. 👍  I'd remove the unnecessary newlines though, let's tighten up the functions.

Reply:
Chatted IRL as the other comment can be done, this will stay the same